staging_vespalib/src/vespa/vespalib/util/malloc_mmap_guard.h

```diff
@@ -0,0 +1,27 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+#pragma once
+
+#include <thread>
+
+namespace vespalib {
+
+/**
+ * Provides a hint to malloc implementation that all allocations in the scope of this guard
+ * will use mmap directly for allocation larger than the givem limit.
```

Comment:
this is technically not completely true since nesting guards will set the value to -1 prematurely.

Reply:
Correct, I have updated comment with intended use.